src/cert_provider/cert_provider_test.cc

```diff
@@ -0,0 +1,503 @@
+#include <gtest/gtest.h>
+
+#include <boost/asio/io_service.hpp>
+#include <boost/format.hpp>
+#include <boost/process.hpp>
+
+#include "config/config.h"
+#include "utilities/utils.h"
+
+static const char* CERT_PROVIDER_PATH = nullptr;
+static const char* CREDENTIALS_PATH = nullptr;
```

Comment:
We have a fondness for boost::filesystem::path (and descriptive types in general).

Reply:
ok, will update it.